pkg/operator/controller/controller.go

```diff
@@ -156,9 +136,9 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 				if err := r.ensureIngressDeleted(ingress, dnsConfig, infraConfig); err != nil {
 					errs = append(errs, fmt.Errorf("failed to ensure ingress deletion: %v", err))
 				}
-			} else {
+			} else if caSecret != nil {
```

Comment:
We only care about the CA secret as part of ensureDefaultCertificateForIngress and only then in the default case, so it seems very weird to treat it here as a "global invariant" that would prohibit doing the main reconcile... it seems like the factoring is just wrong here. It could be created lazily for the first clusteringress that needs it down in ensureClusterIngress and then looked up again later as needed during publishing.

Reply:
Fixed